web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +212,237 @@ describe("<TaskHeader />", () => {
+          },
+        });
+
+        renderTaskHeader(taxTask, userData);
+        selectCompleted();
+
+        expect(
+          screen.queryByText(markdownToText(Config.profileDefaults.STARTING.businessName.header))
+        ).not.toBeInTheDocument();
+        expect(
+          screen.queryByText(markdownToText(Config.profileDefaults.STARTING.taxId.header))
+        ).not.toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.ownership.header))
+        ).toBeInTheDocument();
+      });
+    });
+
+    describe("when public filing legal structure", () => {
+      it("shows all fields when business name not populated", () => {
+        const taxTask = generateTask({ id: taxTaskId });
+        const userData = generateUserData({
+          profileData: generateProfileData({
+            businessName: "",
+            legalStructureId: randomPublicFilingLegalStructure(),
+          }),
+          taskProgress: {
+            [taxTaskId]: "NOT_STARTED",
+          },
+        });
+
+        renderTaskHeader(taxTask, userData);
+        selectCompleted();
+
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.businessName.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.taxId.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.ownership.header))
+        ).toBeInTheDocument();
+      });
+
+      it("does not show business name field when already populated", () => {
+        const taxTask = generateTask({ id: taxTaskId });
+        const userData = generateUserData({
+          profileData: generateProfileData({
+            businessName: "test-business",
+            legalStructureId: randomPublicFilingLegalStructure(),
+          }),
+          taskProgress: {
+            [taxTaskId]: "NOT_STARTED",
+          },
+        });
+
+        renderTaskHeader(taxTask, userData);
+        selectCompleted();
+
+        expect(
+          screen.queryByText(markdownToText(Config.profileDefaults.STARTING.businessName.header))
+        ).not.toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.taxId.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.existingEmployees.header))
+        ).toBeInTheDocument();
+        expect(
+          screen.getByText(markdownToText(Config.profileDefaults.STARTING.ownership.header))
+        ).toBeInTheDocument();
```

Comment:
I'd say these last 3 assertions are unncessary because we've already asserted them in the previous test